Islandora-CLAW/islandora
# Abstract derivative events and bring image/video in-house

**GitHub Issue**: https://github.com/Islandora-CLAW/CLAW/issues/956

Related: https://github.com/Islandora-CLAW/islandora_demo/pull/6

# What does this Pull Request do?

Brings islandora_image inside as a small sub-module and adds islandora_video as well.

# What's new?

Makes an AbstractGenerateDerivative class which both image and video override to generate their separate events.

* Does this change require documentation to be updated? Some is included.
* Does this change add any new dependencies? no
* Does this change require any other modifications to be made to the repository
 (ie. Regeneration activity, etc.)? no
* Could this change impact execution of existing code? no

# How should this be tested?

You can `dsm()` to see the events but this will tie in to issues https://github.com/Islandora-CLAW/CLAW/issues/943 and https://github.com/Islandora-CLAW/CLAW/issues/957 to generate video derivatives.

Currently images should still work, though we will need a claw-playbook PR to remove islandora_image from being installed.

# Interested parties
@Islandora-CLAW/committers @Natkeeran (specifically)
